compiler/z/codegen/ControlFlowEvaluator.cpp

```diff
@@ -893,19 +893,63 @@ lcmpHelper(TR::Node * node, TR::CodeGenerator * cg)
    return targetRegister;
    }
 
+static TR::Register * maxMinHelper(TR::Node *node, TR::CodeGenerator *cg, bool isMax)
+   {
+   TR::Register *registerA = cg->gprClobberEvaluate(node->getFirstChild());
+   TR::Register *registerB = cg->evaluate(node->getSecondChild());
+   // Mask is 4 to pick b when a is Lower for max, 2 to pick b when a is higher for min
+   uint8_t mask = isMax ? 0x4 : 0x2;
+
+   if (node->getOpCodeValue() == TR::imax || node->getOpCodeValue() == TR::imin)
+      {
+      generateRRInstruction(cg, TR::InstOpCode::CR, node, registerA, registerB);
+      generateRRFInstruction(cg, TR::InstOpCode::LOCR, node, registerA, registerB, mask, true);
+      }
+   else if (node->getOpCodeValue() == TR::lmax || node->getOpCodeValue() == TR::lmin)
```

Comment:
I'd like to see an `else` with a `TR_ASSERT_FATAL` in case someone tries to use this helper for `bmax` for example (an opcode that is not supported by this function).

Reply:
6080fa1